lib/routes/recovery-key.js

```diff
@@ -0,0 +1,190 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const errors = require('../error')
+const validators = require('./validators')
+const isA = require('joi')
+const butil = require('../crypto/butil')
+
+module.exports = (log, db, Password, verifierVersion, customs) => {
+  return [
+    {
+      method: 'POST',
+      path: '/recoveryKey',
+      options: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            recoveryKeyId: validators.recoveryKeyId,
+            recoveryData: validators.recoveryData
+          }
+        }
+      },
+      handler: async function (request) {
+        log.begin('createRecoveryKey', request)
+
+        const uid = request.auth.credentials.uid
+        const sessionToken = request.auth.credentials
+        const {recoveryKeyId, recoveryData} = request.payload
+
+        return createRecoveryKey()
+          .then(emitMetrics)
+          .then(() => { return {} })
+
+        function createRecoveryKey() {
+          if (sessionToken.tokenVerificationId) {
+            throw errors.unverifiedSession()
+          }
```

Comment:
Not for this PR, but have enough routes with this logic now, that's it's probably worth factoring out into a "verifiedSessionToken" auth strategy.

Reply:
Made bug to track [this](https://github.com/mozilla/fxa-auth-server/issues/2528).